### PR TITLE
Factor function and color_hue mapper for discrete_fill/color_scale from python to kotlin

### PR DESCRIPTION
**Pull request by IKupriyanov-HORIS**

### CODE REVIEW

File: plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            val mapping = layer.section(Layer.MAPPING)!!.entries.associateBy({ it.value }, { it.key as String })
+
+            return layer.sections(DATA_META, SeriesAnnotation.TAG)!!
+                .map { varMeta -> Pair(varMeta.read(NAME), varMeta.read(CATEGORY))}
+                .filter { (varName, _) -> varName in mapping }
+                .mapNotNull { (varName, category) ->
+                    val aesOption = mapping.getValue(varName)
+                    when (toAes(aesOption)) {
+                        Aes.X, Aes.Y -> mutableMapOf<String, Any>()
+                        Aes.COLOR, Aes.FILL -> mutableMapOf<String, Any>(Scale.SCALE_MAPPER_KIND to "color_hue")
```
Comment (alshan) [inline plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt]: Color/fill mapper_kind default is done in ScaleConfig.kt

File: plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                        else -> null
+                    }?.let { scaleSpec ->
+                        scaleSpec[Scale.AES] = aesOption
```
Comment (alshan) [inline plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt]: Why can't we do this for any AES? 
Is there a constant for "discrete"?

File: plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/config/Option.kt [hunk 3]
```diff
@@ -38,6 +38,12 @@ object Option {
             const val REQUEST = "request"
             const val OSM_ID = "id"
         }
+
+        object SeriesAnnotation {
+            const val TAG = "series_annotation"
+            const val NAME = "variable"
+            const val CATEGORY = "category"
```
Comment (alshan) [inline plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/config/Option.kt]: May be 
NAME --> VARIABLE
CATEGORY --> ANNOTATION
?

Comment (IKupriyanov-HORIS) [reply plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/config/Option.kt]: Looks fine. Fixed.

File: plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt (hunk 2 above)
Comment (IKupriyanov-HORIS) [reply plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt]: I think there are no reasons to limit AES. Fixed.
I introduced a new constant into SeriesAnnotation.

File: plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt (hunk 1 above)
Comment (IKupriyanov-HORIS) [reply plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/server/config/transform/DiscreteScaleFromMetaChange.kt]: Fixed.

File: plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/config/Option.kt [hunk 4]
```diff
@@ -38,6 +38,13 @@ object Option {
             const val REQUEST = "request"
             const val OSM_ID = "id"
         }
+
+        object SeriesAnnotation {
+            const val TAG = "series_annotation"
+            const val VARIABLE = "variable"
+            const val ANNOTATION = "annotation"
+            const val DISCRETE = "discrete"
```
Comment (alshan) [inline plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/config/Option.kt]: Wait, is this same "discrete" as Option.Scale.DISCRETE_DOMAIN?

Comment (alshan) [reply plot-config-portable/src/commonMain/kotlin/jetbrains/datalore/plot/config/Option.kt]: No, its probably not. sorry.

**Pull request merged**